libs/ml/include/ml/dataloaders/commodity_dataloader.hpp

```diff
@@ -152,6 +170,15 @@ void CommodityDataLoader<LabelType, InputType>::GetAtIndex(CommodityDataLoader::
   buffer_.second = std::vector<InputType>({data_.View(index).Copy()});
 }
 
+template <typename LabelType, typename InputType>
+void CommodityDataLoader<LabelType, InputType>::UpdateCursor()
+{
+  if (this->mode_ != DataLoaderMode::TRAIN)
```

Comment:
might want to double check this with emma - seems like it would be trivial to have a validation support for commodity dataloader